Relative dates (#651)
akeneo/pim-helpcenter: content/md/productivity/get-started-with-the-rules-engine.md

The possible operators for the `created` field are:
- =
- !=

> - \>

Reviewer: nope, `>` has a special meaning in yaml (multiline input), that's why it was surrounded by quotes

Author: ah!! because I was told that it was strange to have "" only for >

- <
- BETWEEN
- NOT BETWEEN
- EMPTY
- NOT EMPTY

::: info
The format of the date is:
- "yyyy-mm-dd HH:MM:SS" (UTC time)
- "now"
- "\<relative date format\>" (see below)

If the operator is EMPTY or NOT EMPTY, the value element will be ignored.
:::

:::warning

> The "relative date format" only works with the `<`, `>`, `=` and `!`

Reviewer: ```suggestion
The "relative date format" only works with the `<`, `>`, `=` and `!=` operators
```

:::

### Example

```YML
- field: created
  operator: =
  value: "2015-01-23 00:00:00"
- field: created
  operator: <
  value: "-10 days"
```

#### Focus on the "relative date format"
The *relative date format* allows to specify dates that are relative to the rule execution date, it is formatted as follows:

**<+/->\<count\> \<unit\>**, with:

- **"+"** means a date in the future, **"-"** a date in the past
- **count** is an integer
- **unit** is one of the following values: *minute*, *hour*, *day*, *week*, *month* or *year* with an optional final *s*

For instance, **+1 month** means *in one month*, and **-2 days** means *2 days ago*

:::warning
Obviously, for the `created` and `updated` properties, the only relevant relative date format is the "past" relative date.
:::

## Updated

The possible operators for the `updated` field are:  
- =
- !=

> - \>

Reviewer: nope

- <
- BETWEEN
- NOT BETWEEN
- EMPTY
- NOT EMPTY

::: info
The format of the date is:
- "yyyy-mm-dd HH:MM:SS" (UTC time)
- "now"
- "\<relative date format\>" (see below)

If the operator is EMPTY or NOT EMPTY, the value element will be ignored.
:::

### Example

```YML
-
  field: updated
  operator: =
  value: "2015-01-23 00:00:00"
-
  field: updated

>   operator: >

Reviewer: ```suggestion
  operator: '>'
```

  value: "-1 year"
```

#### Focus on the "relative date format"
The *relative date format* allows to specify dates that are relative to the rule execution date, it is formatted as follows:

**<+/->\<count\> \<unit\>**, with:

- **"+"** means a date in the future, **"-"** a date in the past
- **count** is an integer
- **unit** is one of the following values: *minute*, *hour*, *day*, *week*, *month* or *year* with an optional final *s*

For instance, **+1 month** means *in one month*, and **-2 days** means *2 days ago*

:::warning

> Obviously, for the `created` and `updated` properties, the only relevant relative date format is the "past" relative date.

Reviewer: do we need this part twice?

The possible operators for the `completeness` field are:  
- =
- !=

> - \>

Reviewer: still no :)

The possible operators for the `Date` attribute type are:  
- <
- '>'
- =
- !=
- BETWEEN
- NOT BETWEEN
- EMPTY
- NOT EMPTY

::: info
The format of the date is:
- "yyyy-mm-dd"
- "now"
- "\<relative date format\>" (see [below](#focus-on-the-relative-date-format-2))

> If the operator is EMPTY or NOT EMPTY, the value element will be ignored.

Reviewer: maybe add a list of operators supported with relative dates (same as created/updated, plus ">=" and "<=")

:::

### Example

```YML
-
  field: release_date
  operator: =
  value: "2015-01-23"
-
  field: creation_date
  operator: '>'
  value: "-1 year"
```

#### Focus on the "relative date format"
The *relative date format* allows to specify dates that are relative to the rule execution date, it is formatted as follows:

**<+/->\<count\> \<unit\>**, with:

- **"+"** means a date in the future, **"-"** a date in the past
- **count** is an integer
- **unit** is one of the following values: *day*, *week*, *month* or *year* with an optional final *s*

For instance, **+1 month** means *in one month*, and **-2 days** means *2 days ago*

### Example

```YML
field: created_date
operator: '>'
value: "2016-05-12"
```

:::warning

> The "relative date format" is based on the UTC timezone. It means that if you are located in Australia (UTC +11) and using the PIM on the 06/22/20 at 5:00pm, the "relative date" will be based on the 06/22/20, at 6:00am.

Reviewer: ```suggestion
The "relative date format" is based on the UTC timezone. It means that if you are located in Eastern Australia (UTC +10) and executing the rules on the 06/22/20 at 8:00am, the "relative date" will be based on the 06/21/20
```
